control/control.go

```diff
@@ -202,7 +188,9 @@ func (c *Controller) Session(stream controlapi.Control_SessionServer) error {
 	logrus.Debugf("session started")
 	conn, opts := grpchijack.Hijack(stream)
 	defer conn.Close()
-	err := c.opt.SessionManager.HandleConn(stream.Context(), conn, opts)
+	// FIXME mw0
```

Comment:
I think `SessionManager` probably needs to remain in the manager and we may need to do some redirect to the actual worker per vertex. Even if we could figure out the worker that needs to load local files, for pull credentials multiple workers would need to use the same session. 

Another idea would be to add some logic in the client so that every session request already goes to a specific worker.